### ISSUE TITLE
unnecessary_find_map false positive

### ISSUE THREAD

**Issue by nrempel** [rust-lang/rust-clippy#8966]
### Summary

Hi there,

I seem to encounter a false positive with the following code. I don't think it's possible to replace this `find_map` with `find`.

Thanks!

CC: @djc 

### Lint Name

unnecessary_find_map

### Reproducer

I tried this code:

```rust
// fn clean(word: &str, query: Option<&str>) -> Option<String> {
    let mut query_found = false;
    let part = word.split(|c: char| !c.is_alphabetic()).find_map(|part| {
        if part.is_empty() {
            None
        } else if part.len() == word.len() {
            Some(part)
        } else if Some(part) == query {
            query_found = true;
            None
        } else {
            Some(part)
        }
    });
```

I saw this happen:

```
warning: this `.find_map` can be written more simply using `.find`
   --> vectors/src/generator.rs:255:16
    |
255 |       let part = word.split(|c: char| !c.is_alphabetic()).find_map(|part| {
    |  ________________^
256 | |         if part.is_empty() {
257 | |             None
258 | |         } else if part.len() == word.len() {
...   |
265 | |         }
266 | |     });
    | |______^
    |
    = note: `#[warn(clippy::unnecessary_find_map)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_find_map

```

### Version

```text
rustc 1.61.0 (fe5b13d68 2022-05-18)
binary: rustc
commit-hash: fe5b13d681f25ee6474be29d748c65adcd91f69e
commit-date: 2022-05-18
host: aarch64-apple-darwin
release: 1.61.0
LLVM version: 14.0.0
```

**Comment by Jarcho** [comment]
You can do:
```rust
    let mut query_found = false;
    let part = word.split(|c: char| !c.is_alphabetic()).find_map(|part| {
        if part.is_empty() {
            false
        } else if part.len() == word.len() {
            true
        } else if Some(part) == query {
            query_found = true;
            false
        } else {
            true
        }
    });
```

**Comment by nrempel** [comment]
Ah, thank you! This seems very obvious with fresh eyes 😅
